feedr-starter-code/js/app.js

```diff
@@ -1,3 +1,351 @@
+//         //var retval = []
+//         $('.article').each(function(){
+//            // console.log(this)
+//             retval.push($(this).find('featuredImage').attr('section'))
+//         })
+//        // return console.log(retval);
+//         //console.log(retval);
+//        
+////        $("#popUp").style.display = 'flex';
+//        
+////         var popup = document.getElementById("#popUp");
+////  popup.classList.toggle(".show");
+//
+// });
+
+    $('.closePopUp').click(function(event){
+                    console.log("x clicked");
+        $("#popUp").addClass( "loader hidden" );
+
+    })
+    
+//    $("#list1").on('click', function(event) {
+//  event.preventDefault();
+//    console.log("list 1 clicked");
+//
+//});
+    
+});
+
+function getDiggData(){
+           console.log("in");
+
+     $.get('https://accesscontrolalloworiginall.herokuapp.com/http://digg.com/api/news/popular.json')
+     .then(response => {
+         var feedArray = response.data.feed;
+         fillArticles(feedArray);
+             //   console.log(response.data.feed);
+
+//         feedArray.forEach(function(element) {
+//            console.log(element.content.title);
+//            console.log(element.content.domain_name);
+//            console.log(element.content.description);
+//            console.log(element.content.media.images[3].url);
+
+         
+     });
+}
+    console.log(indexArray.length);
+
+//news api url need to be update every day
+ function getNewsApiData(){
+             console.log("in newsApi");
+
+     $.get('https://accesscontrolalloworiginall.herokuapp.com/https://newsapi.org/v2/everything?q=bitcoin&sortBy=publishedAt&apiKey=e75626217a1249919184dd5d7679e3e7')
+     .then(response => {
+                     // console.log(response);
+
+         var feedArray = response.articles;
+                      console.log(feedArray);
+
+        // fillArticles(feedArray);
+                const arrayOfjQueryArticles = [];
+
+feedArray.forEach((article, index) => {
+
+  let title = article.title;
+    //console.log(title);
+var defaultImg = "http://www.himalayansolution.com/img/medium-default-product.jpg"
+  let imagSrc = article.urlToImage;
+    let imgSrc = imagSrc ? imagSrc :defaultImg
+    let articleURL = article.url;
+    //console.log(articleURL)
+    let category = "news";
+    let source = article.source.name;
+//  if (imgSrc === ""){
+//      imgSrc = "http://www.himalayansolution.com/img/medium-default-product.jpg";
+//  }
+  
+ //$('#no').text(title)
+ let newArticle = fillOneArticle(title ,imgSrc , articleURL, category, source, index);
+
+  arrayOfjQueryArticles.push(newArticle);
+    $('#main').append(newArticle);
+    indexArray.push(newArticle);
+    
+     //   arrayOfjQueryArticles = [] 
+});
+                            $('.article').on('click', function() { 
+         
+        console.log("article clicked!!!!!!!");
+         var indexArray = $(this).attr('id');
```

Comment:
It's better if you change the name of variable here because it's already declared as a global variable.
Or you can reassign.
```suggestion
	  indexArray = $(this).attr('id');
```